### DISCUSSION TITLE
Can Octal SPI on the ESP32-S2/S3 be used to drive a parallel port interface?

### DISCUSSION THREAD

**Discussion by XanCraft21** [espressif/arduino-esp32#7787] [General]
Hello, random idea, is it possible to use Octal SPI to drive a parallel port device like a lcd display? I had an idea to try it and i wanted to ask if other people have tried it yet. Looking at the official documentation about the SPI bus on the ESP32-S2 and the ESP32-S3, both support Octal SPI which I believe is 8 data lines. My theory is i can use this to drive a parallel only tft lcd display with great speed and efficiency without any special external hardware. Any answers or recommendations are good. Thank you.

**Comment by me-no-dev** [comment]
S2 and S3 actually have a dedicated peripheral to drive such displays and supports 16 bits/lanes as well. Check out https://github.com/espressif/arduino-esp32/tree/master/tools/sdk/esp32s3/include/esp_lcd

**Comment by XanCraft21** [comment]
Thank you. I looked at the code and i do not know which display driver chip it is designed to drive. My project requires a specific type of display. The reason why i asked about octal SPI is because i wanted to design a universal display library for a specific bus interface. I plan to check out the pre-made drivers too, as well as experiment with octal SPI.

Do you have any example code for Arduino using the dedicated SPI drivers with the ESP-IDF?

**Comment by Jason2866** [comment]
@XanCraft21 Project Tasmota has implemented a universal Display driver. Needs just an display descriptor file and a lot of display controllers and displays can be used. Some Infos [here](https://tasmota.github.io/docs/Displays/#universal-display-driver) Code is based on Arduino
[Example](https://github.com/arendst/Tasmota/blob/development/tasmota/displaydesc/MF_ILI9488_p16_display.ini) display descriptor to init a parallel RGB display The lib is [here](https://github.com/arendst/Tasmota/tree/development/lib/lib_display/UDisplay)

**Comment by XanCraft21** [comment]
Ok, i will check that out soon. Thank you again.
